base/linalg/bidiag.jl

```diff
@@ -109,6 +109,44 @@ ctranspose(M::Bidiagonal) = Bidiagonal(conj(M.dv), conj(M.ev), !M.isupper)
 istriu(M::Bidiagonal) = M.isupper || all(M.ev .== 0)
 istril(M::Bidiagonal) = !M.isupper || all(M.ev .== 0)
 
+function tril(M::Bidiagonal, k::Integer=0)
+    n = length(M.dv)
+    if abs(k) > n
+        throw(ArgumentError("requested diagonal, $k, out of bounds in matrix of size ($n,$n)"))
+    elseif M.isupper && k < 0
```

Comment:
Maybe this could be:
```julia
    elseif k > 0
          return M
    elseif M.isupper
          if k < 0
               return Bidiagonal(zeros(M.dv),zeros(M.ev),true)
          else
               return Bidiagonal(M.dv,zeros(M.ev),true)
          end
    elseif k < -1
       return Bidiagonal(zeros(M.dv),zeros(M.ev),false)
    elseif k == -1
       return Bidiagonal(zeros(M.dv),M.ev,false)
    else
       return M
    end
```
This avoids the rechecking of M.isupper and passing it to Bidiagonal.